src/EnergyPlus/PurchasedAirManager.cc

```diff
@@ -233,876 +233,953 @@ void GetPurchasedAir(EnergyPlusData &state)
-                    ShowContinueError(state,
-                                      format("{} does not appear in an OutdoorAir:NodeList or as an OutdoorAir:Node.", s_ipsc->cAlphaArgs(13)));
-                    ShowContinueError(state, format("Adding OutdoorAir:Node={}", s_ipsc->cAlphaArgs(13)));
+                    ShowWarningError(state, format("{}{}=\"{} missing data", RoutineName, s_ipsc->cCurrentModuleObject, PurchAir.Name));
+                    ShowContinueError(state, format("{} does not appear in an OutdoorAir:NodeList or as an OutdoorAir:Node.", oaInletNodeName));
+                    ShowContinueError(state, format("Adding OutdoorAir:Node={}", oaInletNodeName));
                 }
                 UniqueNodeError = false;
-                CheckUniqueNodeNames(state, s_ipsc->cAlphaFieldNames(13), UniqueNodeError, s_ipsc->cAlphaArgs(13), s_ipsc->cAlphaArgs(1));
+                CheckUniqueNodeNames(state, cAlphaFieldName, UniqueNodeError, oaInletNodeName, PurchAir.Name);
                 if (UniqueNodeError) {
                     ErrorsFound = true;
                 }
-
                 // get Demand controlled ventilation type
-                PurchAir(PurchAirNum).DCVType = static_cast<DCV>(getEnumValue(dcvNamesUC, s_ipsc->cAlphaArgs(14)));
-                if (PurchAir(PurchAirNum).DCVType == DCV::Invalid) {
-                    ShowSevereInvalidKey(state,
-                                         eoh,
-                                         s_ipsc->cAlphaFieldNames(14),
-                                         s_ipsc->cAlphaArgs(14),
-                                         "Valid entries are None, OccupancySchedule, or CO2Setpoint");
+                cAlphaFieldName = "Demand Controlled Ventilation Type";
+                fieldValue = s_ip->getAlphaFieldValue(fields, schemaProps, "demand_controlled_ventilation_type");
+                PurchAir.DCVType = static_cast<DCV>(getEnumValue(dcvNamesUC, Util::makeUPPER(fieldValue)));
+                if (PurchAir.DCVType == DCV::CO2SetPoint) {
+                    if (!state.dataContaminantBalance->Contaminant.CO2Simulation) {
+                        PurchAir.DCVType = DCV::None;
+                        ShowWarningError(state, format("{}{}=\"{} invalid data", RoutineName, s_ipsc->cCurrentModuleObject, PurchAir.Name));
+                        ShowContinueError(state, format("{}={} but CO2 simulation is not active.", cAlphaFieldName, fieldValue));
+                        ShowContinueError(state, format("Resetting {} to NoDCV", cAlphaFieldName));
+                        ShowContinueError(state,
+                                          "To activate CO2 simulation, use ZoneAirContaminantBalance object and specify \"Carbon Dioxide "
+                                          "Concentration\"=\"Yes\".");
+                    }
+                } else if (PurchAir.DCVType == DCV::Invalid) {
```

Comment:
This `else if` is not possible since there is a default. The else if can be deleted (it will never execute).

    A14, \field Demand Controlled Ventilation Type
       \key None
       \key OccupancySchedule
       \key CO2Setpoint
       \default None